pkg/workloadmanager/sandbox_controller.go

```diff
@@ -75,5 +85,15 @@ func (r *SandboxReconciler) WatchSandboxOnce(_ context.Context, namespace, name
 	fmt.Printf("Registered for future notification for sandbox %s/%s\n", key.Namespace, key.Name)
 	r.mu.Unlock()
 
-	return resultChan
+	// Return cleanup function that removes the entry from pendingRequests
+	cleanup := func() {
```

Comment:
why donot you provide a `UnWatchSandbox` method instead of returning the cleanup